lib/awful/tooltip.lua

```diff
@@ -83,7 +86,87 @@ local offset = {
     bottom       = {x =  0, y =  1 },
 }
 
--- Place the tooltip under the mouse.
+--- The tooltip border color.
+-- @beautiful beautiful.tooltip_border_color
+
+--- The tooltip background color.
+-- @beautiful beautiful.tooltip_bg
+
+--- The tooltip foregound (text) color.
+-- @beautiful beautiful.tooltip_fg
+
+--- The tooltip font.
+-- @beautiful beautiful.tooltip_font
+
+--- The tooltip border width.
+-- @beautiful beautiful.tooltip_border_width
+
+--- The tooltip opacity.
+-- @beautiful beautiful.tooltip_opacity
+
+--- The default tooltip shape.
+-- By default, all tooltips are rectangles, however, by setting this variables,
+-- they can default to rounded rectangle or stretched octogons.
+-- @beautiful beautiful.tooltip_shape
+-- @tparam[opt=gears.shape.rectangle] function shape A `gears.shape` compatible function
+-- @see shape
+-- @see gears.shape
+
+local function apply_shape(self)
+    local s = self._private.shape
+
+    if not s then return end
```

Comment:
I think this has to do things like `self.wibox.shape_bounding = nil` and and `self.wibox:set_bg_image(nil)` (or something like this, I'm not completely sure).

Put differently: Does unsetting the shape work correctly?